tests/unittest/llmapi/apps/_test_trtllm_serve_yaml_config.py

```
@@ -0,0 +1,92 @@
+    """
+    Verify that top-level YAML config (max_num_tokens, max_batch_size, max_seq_len)
+    is correctly applied to trtllm-serve.
+
+    This test explicitly verifies the runtime config values match the YAML file,
+    ensuring YAML config is the single source of truth without needing CLI flags.
+
+    Related: nvbug 5692109
+    """
+    # Query the /config endpoint to get actual runtime values
+    config_url = server.url_for("config")
+    response = requests.get(config_url)
```

Ruff (0.14.5)

Probable use of `requests` call without timeout

(S113)